sdk/spring/azure-spring-boot/src/test/java/com/azure/spring/aad/webapi/AADResourceServerOboConfigurationTest.java

```diff
@@ -0,0 +1,96 @@
+import org.springframework.boot.test.context.runner.WebApplicationContextRunner;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.client.registration.InMemoryClientRegistrationRepository;
+import org.springframework.security.oauth2.client.web.OAuth2AuthorizedClientRepository;
+import org.springframework.security.oauth2.client.web.OAuth2LoginAuthenticationFilter;
+import org.springframework.security.oauth2.server.resource.BearerTokenAuthenticationToken;
+
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class AADResourceServerOboConfigurationTest {
+
+    private final WebApplicationContextRunner contextRunner = new WebApplicationContextRunner()
+        .withPropertyValues(
+            "azure.activedirectory.tenant-id=fake-tenant-id",
+            "azure.activedirectory.client-id=fake-client-id",
+            "azure.activedirectory.client-secret=fake-client-secret");
+
+    @Test
+    public void testNotExistBearerTokenAuthenticationToken() {
+        this.contextRunner
+            .withUserConfiguration(AADResourceServerOboConfiguration.class)
+            .withClassLoader(new FilteredClassLoader(BearerTokenAuthenticationToken.class))
+            .run(context -> {
+                assertThat(context).doesNotHaveBean("AADOAuth2OboAuthorizedClientRepository");
+            });
+    }
+
+    @Test
+    public void testNotExistOAuth2LoginAuthenticationFilter() {
+        this.contextRunner
+            .withUserConfiguration(AADResourceServerOboConfiguration.class)
+            .withClassLoader(new FilteredClassLoader(OAuth2LoginAuthenticationFilter.class))
+            .run(context -> {
+                assertThat(context).doesNotHaveBean("AADOAuth2OboAuthorizedClientRepository");
+            });
+    }
+
+    @Test
+    public void testOnlyGraphClient() {
+        this.contextRunner
+            .withUserConfiguration(AADResourceServerOboConfiguration.class)
+            .withPropertyValues("azure.activedirectory.authorization.graph.scopes=https://graph.microsoft.com/User"
+                + ".Read")
+            .run(context -> {
+                final InMemoryClientRegistrationRepository oboRepo = context.getBean(
+                    InMemoryClientRegistrationRepository.class);
+                final OAuth2AuthorizedClientRepository aadOboRepo = context.getBean(
+                    AADOAuth2OboAuthorizedClientRepository.class);
+
+                ClientRegistration graph = oboRepo.findByRegistrationId("graph");
+                Set<String> graphScopes = graph.getScopes();
+
+                assertThat(aadOboRepo).isNotNull();
+                assertThat(oboRepo).isExactlyInstanceOf(InMemoryClientRegistrationRepository.class);
+                assertThat(graph).isNotNull();
+                assertThat(graphScopes).containsOnly("https://graph.microsoft.com/User.Read");
```

Comment:
@saragluna,

> But how do we achieve that in the unit test?

We can do like this:
```
assertThat(result, Matchers.arrayContaining(expected));
```

Refs: https://stackoverflow.com/questions/54439629/hamcrest-matcher-to-compare-two-arrays

Reply:
get it.

Reply:
@chenrujun I see what you're trying to say here. If we have no special logic here and it's just spring doing the parsing, so the array of one element or two elements doesn't make that much difference to me. But we could do with two elements here.